quantum/quantum.c

```diff
@@ -1041,33 +1041,140 @@ void matrix_scan_quantum() {
+#     define AUDIO_BACKLIGHT_ERROR
+#   endif
+#   define TIMER_UNIT 1
+#   define OC_UNIT C
+# elif BACKLIGHT_PIN == B6
+#   ifdef B6_AUDIO
+#     define AUDIO_BACKLIGHT_ERROR
+#   endif
+#   define TIMER_UNIT 1
+#   define OC_UNIT B
+# elif BACKLIGHT_PIN == B5
+#   ifdef B5_AUDIO
+#     define AUDIO_BACKLIGHT_ERROR
+#   endif
+#   define TIMER_UNIT 1
+#   define OC_UNIT A
+# elif BACKLIGHT_PIN == C6
+#   ifdef C6_AUDIO
+#     define AUDIO_BACKLIGHT_ERROR
+#   endif
+#   define TIMER_UNIT 3
+#   define OC_UNIT A
+# endif
+#elif defined(__AVR_ATmega32U2__) || defined(__AVR_ATmega16U2__)
+// PC5 (OC1B), PC6 (OC1A), PB7 (OC1C)
+# if BACKLIGHT_PIN == B7
+#   ifdef B7_AUDIO
+#     define AUDIO_BACKLIGHT_ERROR
+#   endif
+#   define TIMER_UNIT 1
+#   define OC_UNIT C
+# elif BACKLIGHT_PIN == C6
+#   ifdef C6_AUDIO
+#     define AUDIO_BACKLIGHT_ERROR
+#   endif
+#   define TIMER_UNIT 1
+#   define OC_UNIT A
+# elif BACKLIGHT_PIN == C5
+#   ifdef C5_AUDIO
+#     define AUDIO_BACKLIGHT_ERROR
+#   endif
+#   define TIMER_UNIT 1
+#   define OC_UNIT B
+# endif
+#elif defined(__AVR_AT90USB1286__) || defined(__AVR_AT90USB646__)
+// PB5 (OC1A), PB6 (OC1B), PB7 (OC1C)
+// PC4 (OC3C), PC5 (OC3B), PC6 (OC3A
+# if BACKLIGHT_PIN == B7
+#   ifdef B7_AUDIO
+#     define AUDIO_BACKLIGHT_ERROR
+#   endif
+#   define TIMER_UNIT 1
+#   define OC_UNIT C
+# elif BACKLIGHT_PIN == B6
+#   ifdef B6_AUDIO
+#     define AUDIO_BACKLIGHT_ERROR
+#   endif
+#   define TIMER_UNIT 1
+#   define OC_UNIT B
+# elif BACKLIGHT_PIN == B5
+#   ifdef B5_AUDIO
+#     define AUDIO_BACKLIGHT_ERROR
+#   endif
+#   define TIMER_UNIT 1
+#   define OC_UNIT A
+# elif BACKLIGHT_PIN == C6
+#   ifdef C6_AUDIO
+#     define AUDIO_BACKLIGHT_ERROR
+#   endif
+#   define TIMER_UNIT 3
+#   define OC_UNIT A
+# elif BACKLIGHT_PIN == C5
+#   ifdef C5_AUDIO
+#     define AUDIO_BACKLIGHT_ERROR
+#   endif
+#   define TIMER_UNIT 3
+#   define OC_UNIT B
+# elif BACKLIGHT_PIN == C4
+#   ifdef C4_AUDIO
+#     define AUDIO_BACKLIGHT_ERROR
+#   endif
+#   define TIMER_UNIT 3
+#   define OC_UNIT C
+# endif
+#elif defined(__AVR_ATmega32A__)
+// PD4 (OC1B), PD5 (OC1A)
+# if BACKLIGHT_PIN == D4
+#   define TIMER_UNIT 1
+#   define OC_UNIT B
+# elif BACKLIGHT_PIN == D5
+#   define TIMER_UNIT 1
+#   define OC_UNIT A
+# endif
+#endif
+
+#if !defined(TIMER_UNIT) && !defined(OC_UNIT)
+// BACKLIGHT_PIN is not an hardware PWM pin
+#define NO_HARDWARE_PWM
+#endif
+
+#ifndef NO_HARDWARE_PWM
+
+// We need to ensure that the backlight pin is not in use for audio
+#ifdef AUDIO_BACKLIGHT_ERROR
+# error "Incorrect configuration: backlight pin cannot be used for audio"
+#endif
+
+// hackish pre-processor magic to produce correct TCCRxA, TCCRxB, COMxx1
+// OCRxx and ICRx from the defined TIMER_UNIT and OC_UNIT of the current MCU
+#define PASTE_TCCRxA(timer_unit) TCCR##timer_unit##A
+#define EVAL_TCCRxA(unit) PASTE_TCCRxA(unit)
+#define TCCRxA EVAL_TCCRxA(TIMER_UNIT)
+
+#define PASTE_TCCRxB(timer_unit) TCCR##timer_unit##A
```

Comment:
```suggestion
#define PASTE_TCCRxB(timer_unit) TCCR##timer_unit##B
```
It didn't work on my ATMega32A until I made this change.

Reply:
Hey good catch!

Reply:
I know, but I prefer to push a rebased version, I also edited the commit message.